### PR TITLE
multi-namespace: Adding documentation

### PR DESCRIPTION
**Pull request by draychev**
This PR adds `docs/multiple-namespaces.md`, which is documentation on how to enable multiple namespace support.

I propose we leave the Helm templates as they are -- defaulted to observing namespace `default` only.  This is the safe choice for the time being - would not surprise existing customers.
It would take another step, modifying the value.yaml file, to enable more or all namespaces.

The documentation here is missing a section on troubleshooting. (We need a general troubleshooting section).  I'd like to add that after we have an easy way to enable verbose logging -- coming w/ the following PR:  https://github.com/Azure/application-gateway-kubernetes-ingress/pull/292

### CODE REVIEW

File: docs/multiple-namespaces.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+As of version 0.7 [Azure Application Gateway Kubernetes
+Ingress Controller](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/master/README.md) (AGIC) can ingest events from and observe multiple namespaces.
+Should the AKS administrator decide to use [App Gateway](https://azure.microsoft.com/en-us/services/application-gateway/) as an ingress, all
+namespaces will use the same instance of App Gateway. A single installation of
+Ingress Controller will monitor accessible namespaces and will configure the App
+Gateway it is associated with.
+
+#### Defaults
+By default, as of version 0.7, AGIC is configured to observe ingress resources
+from **default** namespace only. This is configured in
```
Comment (akshaysngupta) [inline docs/multiple-namespaces.md]: AGIC is configured to observe ingress resources from **default** namespace or namespace specified in the helm config.

File: docs/multiple-namespaces.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+[values.yaml](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/d213b8803c302dd003ec07f8765faa70ad2ee43e/helm/ingress-azure/values.yaml#L12).
+
+#### Enable multiple namespace support
+To enable miltiple namespace support - modify the
+[values.yaml](https://github.com/Azure/application-gateway-kubernetes-ingress/blob/d213b8803c302dd003ec07f8765faa70ad2ee43e/helm/ingress-azure/values.yaml#L12) file in one of the following ways:
```
Comment (akshaysngupta) [inline docs/multiple-namespaces.md]: You can approach this by maintaining a helm-config.yaml with the watchNamespace.
Then just use, `helm install -f helm-config.yaml application-gateway-kubernetes-ingress/ingress-azure`

Comment (draychev) [reply docs/multiple-namespaces.md]: Yes! Thank you!!

**Comment by draychev** [comment]
@akshaysngupta PTAL

File: docs/multiple-namespaces.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+installation of Ingress Controller will monitor accessible namespaces and will
+configure the App Gateway it is associated with.
+
+#### Defaults
+As of version 0.7, AGIC is configured to observe ingress resources from the
+*default* namespace or the namespace explicitly defined in the Helm
+configuration.
+
+#### Enable multiple namespace support
+To enable miltiple namespace support - modify the [helm-config.yaml](examples/helm-config.yaml) file in one of the following ways:
```
Comment (akshaysngupta) [inline docs/multiple-namespaces.md]: s/miltiple/multiple

File: docs/multiple-namespaces.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+Install Helm template changes with: `helm install -f helm-config.yaml application-gateway-kubernetes-ingress/ingress-azure`
+
+Once deployed with the ability to observe multiple namespaces, AGIC will:
+  - list ingress resources from all accessible namespaces
+  - filter to ingress resources annotated with `kubernetes.io/ingress.class: azure/application-gateway`
+  - compose combined [App Gateway config](https://github.com/Azure/azure-sdk-for-go/blob/37f3f4162dfce955ef5225ead57216cf8c1b2c70/services/network/mgmt/2016-06-01/network/models.go#L1710-L1744)
+  - apply the config to the associated App Gateway via [ARM](https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-overview)
+
+#### Conflicting Configurations
```
Comment (akshaysngupta) [inline docs/multiple-namespaces.md]: As we are mentioning the conflicting scenario, should we mention what two ingresses can share without having conflicts ?

Comment (draychev) [reply docs/multiple-namespaces.md]: Rewrote pieces of that to add clarity.

**Comment by draychev** [comment]
@rheaj99 @akshaysngupta  PTAL -- https://github.com/Azure/application-gateway-kubernetes-ingress/pull/293/files?short_path=83d9ece#diff-83d9ece0617ecf48bfb9cd3687530a6d

I do intend on adding a `Troubleshooting` section but perhaps in another PR.

File: docs/multiple-namespaces.md [hunk 5]
```diff
@@ -0,0 +1,120 @@
+# Multiple Namespace Support
+
+#### Motivation
```
Comment (akshaysngupta) [inline docs/multiple-namespaces.md]: can you add a link from feature doc ?

Comment (draychev) [reply docs/multiple-namespaces.md]: Let's do it in another PR?

**Pull request merged**